dashboard/app/controllers/transfers_controller.rb

```diff
@@ -4,71 +4,51 @@ class TransfersController < ApplicationController
+    if stay_enrolled_in_current_section.nil?
+      return head :bad_request
     end
 
-    # This is needed when we eventually allow students to be in multiple sections
-    if params.key?(:current_section_code)
-      current_section_code = params[:current_section_code]
-    else
+    student_ids = params[:student_ids].try(:split, ',').try(:map, &:to_i)
+    if student_ids.nil? || student_ids.empty?
       render json: {
-        error: I18n.t('move_students.section_code_not_entered')
+        error: I18n.t('move_students.student_ids_not_entered')
       }, status: :bad_request
       return
     end
 
+    # Rather than noop, we return a bad_request if the source and destination sections are the same.
     if new_section_code == current_section_code
       render json: {
         error: I18n.t('move_students.section_code_cant_be_current_section')
       }, status: :bad_request
       return
     end
-
-    current_section = Section.find_by_code(current_section_code)
-    unless current_section
+    # Verify the destination section and destination teacher exist (are not soft-deleted).
+    new_section = Section.find_by_code(new_section_code)
+    unless new_section && new_section.user
       render json: {
-        error: I18n.t('move_students.current_section_dne', current_section_code: current_section_code)
+        error: I18n.t('move_students.new_section_dne', new_section_code: new_section_code)
       }, status: :not_found
       return
     end
-
-    if current_section.user != current_user
-      render json: {
-        error: I18n.t('move_students.students_not_yours')
-      }, status: :forbidden
-      return
-    end
-
-    # As of right now, this only applies to transfers to another teacher
-    # When students are allowed to be in multiple sections, this will also be needed
-    # for transfers between the current logged-in teacher
-    if new_section.user == current_user
-      stay_enrolled_in_current_section = false
-    elsif params.key?(:stay_enrolled_in_current_section)
-      stay_enrolled_in_current_section = params[:stay_enrolled_in_current_section] && params[:stay_enrolled_in_current_section] != 'false'
-    else
+    # Verify the source section exists (is not soft-deleted).
+    current_section = Section.find_by_code(current_section_code)
+    unless current_section
       render json: {
-        error: I18n.t('move_students.stay_enrolled_not_entered')
-      }, status: :bad_request
+        error: I18n.t('move_students.current_section_dne', current_section_code: current_section_code)
+      }, status: :not_found
       return
     end
 
-    if params.key?(:student_ids)
-      student_ids = params[:student_ids].split(',').map(&:to_i)
-    else
-      render json: {
-        error: I18n.t('move_students.student_ids_not_entered')
-      }, status: :bad_request
-      return
-    end
+    # TODO(asher): Determine if this should be :manage (currently not granted) instead of :read.
+    authorize! :read, current_section
```

Comment:
Do we also perhaps want to leverage CanCan's `load_and_authorize_resource` instead of explicitly calling `Section.find_by_code` and `authorize!`? Probably not if we want the above specific localized error messages, but just thought I'd ask.